fw/src/mgos_init.c

```diff
@@ -30,9 +30,9 @@ enum mgos_init_result mgos_init(void) {
     return MGOS_INIT_APP_INIT_FAILED;
   }
 
-  LOG(LL_INFO, ("Init done, RAM: %d total, %d free, %d min free",
-                (int) mgos_get_heap_size(), (int) mgos_get_free_heap_size(),
-                (int) mgos_get_min_free_heap_size()));
+  LOG(LL_INFO, ("Init done, RAM: %u total, %u free, %u min free",
```

Comment:
annoyingly, uint32_t is cast to different things on different platforms - sometimes it';s unsigned int but sometimes it's unsigned long so there's no one true format character for it.
so, when integrating, i slightly modified it to cast to `(unsigned long)` and `%ul` to match. `unsigned long` is 64 bits on amd64, so it'll work for the purpose you're using it here as well.

Reply:
well, it ends up being 32-bit of course. on armhf sizeof(int) and sizeof(long) are both 32 bits, some compiler/library authors define it to `unsigned long`, some to `unsigned int`. we definitely have both flavors among our platforms, but i forget which is which... that's the reaosn you see casting to `(int)` or `(unsigned int)` when printing (i find `PRI*` macros are even worse).